store/src/java/com/zimbra/cs/filter/jsieve/EditHeaderExtension.java

```diff
@@ -564,4 +566,35 @@ private boolean matchValue(String regex, String value) {
         Matcher matcher = pattern.matcher(value);
         return matcher.matches();
     }
+
+    static public boolean isTolerableFormedMessage(ZimbraMailAdapter zma, String actionName, MimeMessage mm) {
+        if (zma.getEheParseStatus() == PARSESTATUS.UNKNOWN) {
+            return saveChanges(zma, actionName, mm);
+        } else if (zma.getEheParseStatus() == PARSESTATUS.MIMEMALFORMED) {
+            ZimbraLog.filter.info(actionName + ": Triggering message is malformed MIME");
+            return false;
+        }
+        return true;
+    }
+
+    static public boolean saveChanges(ZimbraMailAdapter zma, String actionName, MimeMessage mm) {
+        String ct = "";
```

Comment:
whats the use of ct? the parse error could be caused because of other malformed header value too other than contentType.